x-pack/plugins/security_solution/server/endpoint/routes/resolver/tree/queries/stats.ts

```diff
@@ -121,26 +186,99 @@ export class StatsQuery {
    * @param client used to make requests to Elasticsearch
    * @param nodes an array of unique IDs representing nodes in a resolver graph
    */
-  async search(client: IScopedClusterClient, nodes: NodeID[]): Promise<Record<string, EventStats>> {
+  async search(
+    client: IScopedClusterClient,
+    nodes: NodeID[],
+    alertsClient: AlertsClient | undefined,
+    includeHits: boolean
+  ): Promise<{ eventStats?: Record<string, EventStats>; alertIds?: string[] }> {
     if (nodes.length <= 0) {
       return {};
     }
 
     const esClient = this.isInternalRequest ? client.asInternalUser : client.asCurrentUser;
+    const alertIndex =
+      typeof this.indexPatterns === 'string' ? this.indexPatterns : this.indexPatterns.join(',');
 
-    // leaving unknown here because we don't actually need the hits part of the body
-    const body = await esClient.search({
-      body: this.query(nodes),
-      index: this.indexPatterns,
+    const [body, alertsBody] = await Promise.all([
+      await esClient.search({
+        body: this.query(nodes),
+        index: this.indexPatterns,
+      }),
+      alertsClient
```

Comment:
If you prefer, you can throw this in a function that you call here to avoid the nesting?

```
const getAlertsStats = async (alertsClient) => {
   return alertsClient 
     ? await alertsClient.find(this.alertStatsQuery(nodes, alertIndex, includeHits))
     : Promise.resolve({ hits: { hits: [] } })
}
```
Though the promise.resolve _might_ be unnecessary